tools/building.py

```diff
@@ -220,13 +220,13 @@ def lld_flags_for_executable(external_symbols):
       cmd.append('--growable-table')
 
   if not settings.SIDE_MODULE:
-    # Export these two section start symbols so that we can extract the string
-    # data that they contain.
-    cmd += [
-      '-z', 'stack-size=%s' % settings.STACK_SIZE,
-      '--initial-memory=%d' % settings.INITIAL_MEMORY,
-      '--max-memory=%d' % settings.MAXIMUM_MEMORY,
-    ]
+    cmd += ['-z', 'stack-size=%s' % settings.STACK_SIZE]
+    cmd += ['--max-memory=%d' % settings.MAXIMUM_MEMORY]
+
+    if settings.INITIAL_HEAP != -1:
+      cmd += ['--initial-heap=%d' % settings.INITIAL_HEAP]
+    if settings.INITIAL_MEMORY != -1:
```

Comment:
Can these ever both be set?  If not then maybe this should be `elif`?

Reply:
Both can be set, e. g. in the test we do `'-sINITIAL_HEAP=128KB', '-sINITIAL_MEMORY=20MB'`. The behavior is defined by `wasm-ld` in this case.